solution/index.js

```diff
@@ -0,0 +1,541 @@
+    newTaskInput.addEventListener("focus", focusOnNewTaskInput)
+}
+
+
+// Handelers
+function hoverHandler({target}){
+    if (target.classList.contains("task")){
+        selectedTask = target
+        document.addEventListener("keydown", moveBetweenSections)
+        document.addEventListener("dblclick", editTask)
+        selectedTask.addEventListener("blur", finishedit)
+    }
+    target.onmouseout = () => {
+        document.removeEventListener("keypress", moveBetweenSections)
+        selectedTask = null
+    }
+}
+
+            //////////////////////////////////
+            //  features and functionality  //
+            //////////////////////////////////
+
+//  Edit tasks
+function editTask(){
+    if (selectedTask){
+        selectedTask.setAttribute("contenteditable", "true")
+        selectedTask.classList.add("editable")
+        removeTaskData(selectedTask.innerText)
+        updateLocalSaveData()
+        document.addEventListener("keydown", enterToBlur)
+    }
+}
+function finishedit(){  // Saves the new task in local storage
+    for (const task of document.querySelectorAll(".editable")){
+        task.removeAttribute("contenteditable")
+        task.classList.remove("editable")
+        const editedTaskUl = task.parentElement
+        addNewTaskData(editedTaskUl , task.innerText)
+        updateLocalSaveData()
+        document.removeEventListener("keydown", enterToBlur)
+    }
+}
+function enterToBlur(event){
+    if (event.key === "Enter") {
+        event.preventDefault()
+        selectedTask.blur()
+    }
+}
+
+//  Search tasks
+function search({key}){
+    let searchQuery
+    const searchBarEntry = document.getElementById("search").value.toLowerCase()
+    if (key == "Backspace"){
+        searchQuery = searchBarEntry.slice(0, searchBarEntry.length-1)
+    }
+    if (key == "Space") {
+        searchQuery = searchBarEntry+" "
+    }
+    if (key.length === 1){
+        searchQuery = searchBarEntry+key
+    }
+    const tasksList = document.querySelectorAll(".task")
+    if (searchQuery !== undefined){
+        for (const task of tasksList) {
+            if (!(task.innerText.includes(searchQuery))){
+                task.classList.add("invisible")
+            }
+            else {
+                task.classList.remove("invisible")
+            }
+        }
+    }
+}
+
+// API storage
+async function storeTasks(){
+    const loader = createElement("div", [], [], {id: "loading"})
+    const header = document.getElementById("header")
+    header.appendChild(loader)
+    const tasks = taskObj
+    const pushToCloud = await fetch("https://json-bins.herokuapp.com/bin/614ad8b24021ac0e6c080c0b", {
+        method: "PUT",
+        headers: {
+            Accept: "application/json", 
+            "Content-Type": "application/json",
+        },
+        body: JSON.stringify({tasks})
+    })
+    if (!pushToCloud.ok) {
+        alert(`Failed! Error ${response.status}: ${response.statusText}`);
```

Comment:
"storeTasks" function encountered an error, use throw to indicate the problem to caller